### PR TITLE
IEEE-132 and IEEE-133: Add cancelOrderThunk and connect to Order Tables

### PR DESCRIPTION
**Pull request by Mustaballer**
## Overview

- Resolves #IEEE-132 and #IEEE-133
- Create cancelOrderThunk in orderSlice.ts to change order status to "Cancelled"
  - call PATCH request to endpoint: /api/hardware/orders/<id>
  - use removeOne from EntityAdapter
- Connect thunk to Order Tables
  - when cancel button is clicked, it dispatches thunk

## Unit Tests Created

- Tested state of cancelOrderLoadingSelector
- Mocked API Response for cancelOrderThunk and ensured pendingOrderAdapter removed the order 
- Tested error and success snack bars
- Tested is cancel order button is hidden when 'hide all' is selected
- Tested UI when cancel order button is clicked

## Steps to QA

- Add cart items, and submit them as an order. Then go to Dashboard and click 'Cancel Order' to see the orders get removed.

### CODE REVIEW

File: hackathon_site/dashboard/frontend/src/api/api.ts [hunk 1]
```diff
@@ -86,3 +86,8 @@ export const post = <T>(uri: string, data?: any) => {
     uri = cleanURI(uri);
     return axios.post(`${SERVER_URL}/${uri}`, data, makeConfig());
 };
+
+export const patch = <T>(uri: string, data?: any) => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/api/api.ts]: This was missing from post function but can you make the function return type `Promise<AxiosResponse<T>>`

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 2]
```diff
@@ -465,6 +467,18 @@ export const PendingTable = () => {
                                 </TableBody>
                             </Table>
                         </TableContainer>
+                        <div className={styles.title}>
+                            <Typography
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: Why do you have an empty Typography component?
if you want to use styles.title you can put it directly on the Button component, no need for an extra div wrapper

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -46,6 +48,10 @@ const Dashboard = () => {
         }
     }, [dispatch, hardwareInOrders]);
 
+    useEffect(() => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx]: why is this necessary?

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 4]
```diff
@@ -56,6 +56,29 @@ export const getTeamOrders = createAsyncThunk<
     }
 });
 
+export const updateOrder = createAsyncThunk<
+    OrderInTable,
+    number,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(`${orderReducerName}/updateOrder`, async (order_id, { rejectWithValue }) => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: change to `orderId` bc we use camelcase in javascript

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 5]
```diff
@@ -56,6 +56,29 @@ export const getTeamOrders = createAsyncThunk<
...
+    try {
+        const data = {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: you don't need the request property here, PATCH requests are partial updates so status in the request body is enough.

Also setting request to "string" would not be a good idea lol, check out the order model in hardware/models.py to learn more about what the request property does.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 6]
```diff
@@ -56,6 +56,29 @@ export const getTeamOrders = createAsyncThunk<
...
+            status: "Cancelled",
+            request: "string",
+        };
+        const response = await patch<APIListResponse<Order>>(
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: Check swagger again, the return type for this api is not an api list response of orders

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 7]
```diff
@@ -56,6 +56,29 @@ export const getTeamOrders = createAsyncThunk<
...
+            `/api/hardware/orders/${order_id}`,
+            data
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: since the request json has only one property, I think you can put it directly in here.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 8]
```diff
@@ -56,6 +56,29 @@ export const getTeamOrders = createAsyncThunk<
     }
 });
 
+export const updateOrder = createAsyncThunk<
+    OrderInTable,
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: The return type for this thunk is not OrderInTable, the return type is whatever you return (which is response.data --> check swagger to figure out what that is)

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 9]
```diff
@@ -89,6 +112,29 @@ const orderSlice = createSlice({
                 payload?.message ||
                 "There was a problem retrieving orders. If this continues please contact hackathon organizers.";
         });
+
+        builder.addCase(updateOrder.pending, (state) => {
+            state.isLoading = true;
+            state.error = null;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: watch out what ur setting as loading and error, these states are for a different thunk (getTeamOrders thunk) so best not to interfere there, create new states for loading & error for cancelling

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 10]
```diff
@@ -89,6 +112,29 @@ const orderSlice = createSlice({
...
+        });
+
+        builder.addCase(updateOrder.fulfilled, (state, { payload }) => {
+            state.isLoading = false;
+            state.error = null;
+            if (payload) {
+                pendingOrderAdapter.updateOne(state, {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: when you cancel an order, it shouldn't show up on the dashboard anymore, you can remove it from the store

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 11]
```diff
@@ -89,6 +112,29 @@ const orderSlice = createSlice({
...
+                    id: payload.id,
+                    changes: payload,
+                });
+            }
+        });
+
+        builder.addCase(updateOrder.rejected, (state, { payload }) => {
+            state.isLoading = false;
+            state.error =
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: Actually, I don't think we need an error state, you can just dispatch a snackbar here with the error message

**Review by YuYing-Liang — CHANGES_REQUESTED** [review]
Good start!
The comments I left are a bit less helpful than usual only because I want you to get used to figuring things out in the code on your own more.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts (hunk 11 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: I can dispatch a snack bar where I catch errors in the thunk, I don't know if I can dispatch in this rejected case.

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx (hunk 2 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: for whatever reason, the button gets rendered to the right underneath the order table when I have the empty wrapper. Otherwise with that class div, its rendered to the left, not sure how to fix it without adding my own css

Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: Not sure how to resolve this formatting with CSS... 
image

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 12]
```diff
@@ -465,6 +467,14 @@ export const PendingTable = () => {
                                 </TableBody>
                             </Table>
                         </TableContainer>
+                        <TableRow className={styles.title} style={{ float: "right" }}>
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: Instead of float right (float is very bad, shamed upon in css community 😂) try wrapping it in a tablecell component aligned right. And if that doesn't work you can do some more googling

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx [hunk 13]
```diff
@@ -44,7 +46,7 @@ const Dashboard = () => {
             dispatch(setFilters({ hardware_ids: hardwareInOrders }));
             dispatch(getHardwareWithFilters());
         }
-    }, [dispatch, hardwareInOrders]);
+    }, [dispatch, hardwareInOrders, pendingOrders]);
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.tsx]: Why do you need this as a dependency here? It shouldn't be required.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 14]
```diff
@@ -56,6 +68,51 @@ export const getTeamOrders = createAsyncThunk<
     }
 });
 
+export interface PatchOrderResponse {
+    id: number;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: I'm pretty sure this is just the Order type, or u can use parts of the order type here, check types.ts

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 15]
```diff
@@ -89,6 +146,29 @@ const orderSlice = createSlice({
...
+                pendingOrderAdapter.removeOne(state, {
+                    payload: payload.id,
+                    type: payload.status,
+                });
+            }
+        });
+
+        builder.addCase(cancelOrderThunk.rejected, (state, { payload }) => {
+            state.isLoading = false;
+            state.cancelOrderError =
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: If ur displaying the snack bar already, you don't need this error state bc it's not being used anywhere else

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 16]
```diff
@@ -465,6 +467,14 @@ export const PendingTable = () => {
...
+                            <Button
+                                onClick={() => cancelOrder(pendingOrder.id)}
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx:483]: This button should be disabled when the thunk is loading

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 17]
```diff
@@ -465,6 +469,22 @@ export const PendingTable = () => {
                                 </TableBody>
                             </Table>
                         </TableContainer>
+                        <TableCell
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: I played around with this and this element doesn't need to be a TableCell, plain old div is fine. You also don't need styles.title, it does nothing for the button.

For your css, nice job googling lol you don't need flexDirection bc its set to row by default and add some margin to the element as well bc its a bit squished under the table.

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 18]
```diff
@@ -465,6 +469,22 @@ export const PendingTable = () => {
...
+                                flexDirection: "row",
+                                justifyContent: "flex-end",
+                            }}
+                        >
+                            <Button
+                                onClick={() => cancelOrder(pendingOrder.id)}
+                                disabled={isCancelOrderLoading}
+                                color="secondary"
+                            >
+                                {isVisible ? "Cancel order" : ""}
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: you don't need this if statement because isVisibile condition is wrapped around the entire component.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts [hunk 19]
```diff
@@ -134,3 +164,74 @@ describe("getTeamOrders Thunk", () => {
...
+    it("Updates the store on API success", async () => {
+        const response = makeMockApiResponse(mockPendingOrders);
+        mockedPatch.mockResolvedValueOnce(response);
+
+        const store = makeStoreWithEntities({
+            pendingOrders: mockPendingOrdersInTable,
+        });
+        store.dispatch(cancelOrderThunk(3));
+        await waitFor(() => {
+            expect(mockedPatch).toHaveBeenCalledWith(`/api/hardware/orders/${3}`, {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts]: you can just put 3 here, you don't need ${3}

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 20]
```diff
@@ -89,6 +119,24 @@ const orderSlice = createSlice({
...
         });
+
+        builder.addCase(cancelOrderThunk.pending, (state) => {
+            state.cancelOrderLoading = true;
+        });
+
+        builder.addCase(cancelOrderThunk.fulfilled, (state, { payload }) => {
+            state.cancelOrderLoading = false;
+            if (payload) {
+                pendingOrderAdapter.removeOne(state, {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: based on redux toolkit documentation, I think you only need the id for this 2nd argument.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 21]
```diff
@@ -56,6 +59,33 @@ export const getTeamOrders = createAsyncThunk<
...
+    async (orderId, { dispatch, rejectWithValue }) => {
+        try {
+            const response = await patch<Order>(`/api/hardware/orders/${orderId}`, {
+                status: "Cancelled",
+            });
+            return response.data;
+        } catch (e: any) {
+            dispatch(
+                displaySnackbar({
+                    message: `There was a problem cancelling orders. If this continues please contact hackathon organizers.`,
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts]: for snackbar message could you include `e.response?.data?.status[0]` if its not null? take a look at hardware/serializers.py OrderChangeSerializer to get to know the other error messages that might come in. I got an error message because you can't cancel an order that's "ready for pickup"

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts [hunk 22]
```diff
@@ -56,6 +59,33 @@ export const getTeamOrders = createAsyncThunk<
...
+    number,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${orderReducerName}/cancelOrderThunk`,
+    async (orderId, { dispatch, rejectWithValue }) => {
+        try {
+            const response = await patch<Order>(`/api/hardware/orders/${orderId}`, {
+                status: "Cancelled",
+            });
+            return response.data;
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.ts:79]: could you display a snackbar when the order is successfully cancelled too, makes removing the table a bit less abrupt lol

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts [hunk 23]
```diff
@@ -134,3 +164,74 @@ describe("getTeamOrders Thunk", () => {
         );
     });
 });
+
+describe("cancelOrderThunk Thunk", () => {
+    it("Updates the store on API success", async () => {
+        const response = makeMockApiResponse(mockPendingOrders);
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts]: makeMockApiResponse is for single values not lists and mockPendingOrders is a list, so maybe do `mockPendingOrders[3]`

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts [hunk 24]
```diff
@@ -134,3 +164,74 @@ describe("getTeamOrders Thunk", () => {
...
+                    {
+                        id: 4,
+                        requested_quantity: 1,
+                    },
+                ],
+            });
+        });
+    });
+
+    it("Updates the store on API failure", async () => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts]: this isn't rlly "updating the store", maybe changing the name of this test with something to do with displaying a snackbar.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts [hunk 25]
```diff
@@ -134,3 +164,74 @@ describe("getTeamOrders Thunk", () => {
...
+        const store = makeStoreWithEntities({
+            pendingOrders: mockPendingOrdersInTable,
+        });
+        store.dispatch(cancelOrderThunk(3));
+        await waitFor(() => {
+            expect(mockedPatch).toHaveBeenCalledWith(`/api/hardware/orders/${3}`, {
+                status: "Cancelled",
+            });
+            expect(pendingOrderSelectors.selectById(store.getState(), 3)).toEqual({
+                id: 3,
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts]: shouldn't this item just not be there anymore? So this should be equal to undefined.

**Review by YuYing-Liang — COMMENTED** [review]
Not bad Mustafa, but your tests make me feel like some things about this ticket are unclear, lmk if you have any more questions!

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx [hunk 26]
```diff
@@ -73,13 +74,17 @@ describe("<PendingTable />", () => {
         const store = makeStoreWithEntities({
             pendingOrders: mockPendingOrdersInTable,
         });
-        const { getByText, queryByText } = render(<PendingTable />, { store });
+        const { getByText, queryByText, getAllByText } = render(<PendingTable />, {
+            store,
+        });
         const button = getByText(/hide all/i);
+        const cancelOrderBtns = getAllByText(/cancel order/i);
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx]: You can use within() like you did with the cart tests to get the exact cart button of that order. It's not good to test clicking all the cancel buttons at once.

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx [hunk 27]
```diff
@@ -73,13 +74,17 @@ describe("<PendingTable />", () => {
         const store = makeStoreWithEntities({
             pendingOrders: mockPendingOrdersInTable,
         });
-        const { getByText, queryByText } = render(<PendingTable />, { store });
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx:76]: Why are you changing the hide button test? You should create a new test for  the cancel button.

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx [hunk 28]
```diff
@@ -465,6 +469,21 @@ export const PendingTable = () => {
...
+                                justifyContent: "flex-end",
+                                marginTop: "10px",
+                            }}
+                        >
+                            <Button
+                                onClick={() => cancelOrder(pendingOrder.id)}
+                                disabled={isCancelOrderLoading}
+                                color="secondary"
+                            >
+                                {"Cancel order"}
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.tsx]: You don't need the brackets or the quotes here

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx [hunk 29]
```diff
@@ -151,6 +155,28 @@ describe("Dashboard Page", () => {
...
+        mockedPatch.mockResolvedValueOnce(response);
+
+        const store = makeStoreWithEntities({
+            pendingOrders: mockPendingOrdersInTable,
+        });
+
+        const { queryAllByText, getByText } = render(<Dashboard />, {
+            store,
+        });
+        const cancelOrderBtns = queryAllByText(/cancel order/i);
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx]: Same thing here, it would be cleaner code if you used within() to get the correct cancel button

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx [hunk 30]
```diff
@@ -151,6 +155,28 @@ describe("Dashboard Page", () => {
...
+        const orderTable = getByText(/order #4/i);
+
+        fireEvent.click(cancelOrderBtns[1]);
+
+        await waitFor(() => {
+            expect(orderTable).not.toBeInTheDocument();
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx]: In addition to making sure the text isn't there, you can also make sure the table isn't in the document either with getByTestId

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx [hunk 31]
```diff
@@ -151,6 +155,28 @@ describe("Dashboard Page", () => {
             expect(mockedGet).toHaveBeenCalledWith("/api/event/teams/team/");
         });
     });
+
+    it("Removes orders when cancel order button is clicked", async () => {
+        const response = makeMockApiResponse(mockPendingOrdersInTable[1]);
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx]: For mocking apis this isn't the best practice because your mocking all patch requests. You should be using when() and specifying the exact api url, check previous api tests for examples.

File: hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts [hunk 32]
```diff
@@ -134,3 +165,62 @@ describe("getTeamOrders Thunk", () => {
...
+                options: { variant: "success" },
+            })
+        );
+    });
+
+    it("displaying snackbar on error", async () => {
+        const failureResponse = {
+            response: {
+                status: 500,
+                message: `Failed to cancel order: undefined`,
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/slices/order/orderSlice.test.ts:209]: Interesting error message lol

**Review by YuYing-Liang — COMMENTED** [review]
Some issues with the tests but otherwise, pretty good! Nice job 😊

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx (hunk 27 above)
Comment (Mustaballer) [reply hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx:76]: I wanted to add to this test for when the hide button is clicked, you wouldn't be able to see the cancel button

File: hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx [hunk 33]
```diff
@@ -82,6 +82,21 @@ describe("<PendingTable />", () => {
...
+        const store = makeStoreWithEntities({
+            pendingOrders: mockPendingOrdersInTable,
+        });
+        const { getByText, queryByText } = render(<PendingTable />, {
+            store,
+        });
+        const button = getByText(/hide all/i);
+        fireEvent.click(button);
+        expect(getByText(/show all/i)).toBeInTheDocument();
+        mockPendingOrders.map(() => {
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/components/dashboard/ItemTable/ItemTable.test.tsx]: if you want to check that cancel order button is not there for any order, you don't need this map because you're not using anything from the array. Instead what this code is doing is running `expect(queryByText(/cancel order/i)).toBeNull();` multiple times, which is unnecessary.

File: hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx [hunk 34]
```diff
@@ -151,6 +155,46 @@ describe("Dashboard Page", () => {
...
+            store,
+        });
+
+        await waitFor(() => {
+            mockPendingOrdersInTable.forEach((order) => {
+                expect(getByText(`Order #${order.id}`)).toBeInTheDocument();
+            });
+        });
+
+        const orderItem = getByTestId(
```
Comment (YuYing-Liang) [inline hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx:180]: just fyi, this data test id is on the order table, not an order item (an order item is a hardware item you put in the order). 
Naming this table "pending-order-item" may cause confusion because in the backend we have an OrderItem table that keeps track of order items which are different from orders.

Comment (YuYing-Liang) [reply hackathon_site/dashboard/frontend/src/pages/Dashboard/Dashboard.test.tsx:180]: To avoid this confusion we may want to rename `<PendingTable/>` to `<PendingTables/>` since all the pending orders are in that component. Same with CheckedOutTable, but that can be a separate ticket.

**Review by YuYing-Liang — COMMENTED** [review]
some last comments, overall good job!

**Review by YuYing-Liang — APPROVED** [review]
thanks for renaming!
Looks good to me :)
